src/analysis/processing/qgsalgorithmremoveholes.cpp

```diff
@@ -111,7 +111,7 @@ QgsFeatureList QgsRemoveHolesAlgorithm::processFeature( const QgsFeature &featur
     if ( mDynamicMinArea )
       minArea = mMinAreaProperty.valueAsDouble( context.expressionContext(), minArea );
 
-    f.setGeometry( geometry.removeInteriorRings( minArea > 0 ? minArea : -1 ) );
+    f.setGeometry( geometry.removeInteriorRings( minArea > 0 ? minArea : -1 ).buffer( 0, 0 ) );
```

Comment:
Unfortunately this will lose any curves, z or m values.

Reply:
You're right. I was just looking at the methods that cause the M to be lost to continue to try to fix this and it is one of them...